### DISCUSSION TITLE
qutebrowser: cannot start mpv

### DISCUSSION THREAD

**Discussion by AblatedSprocket** [netblue30/firejail#5724] [Q&A] [answered]
In Qutebrowser, I have a binding set up to spawn MPV, which works without Firejail. With Firejail, I get this error:

Command exited with status 127.

/usr/bin/mpv: error while loading shared libraries: libluajit-5.1.so.2: cannot open shared object file: Permission denied

I'm assuming this is because Firejail can't see the lib, so my first idea was to try to whitelist it. I found this file in /usr/lib/, so I assumed the easy, not great practice solution would be to whitelist /usr/lib, which it turns out is not a valid whitelist path.

Can someone point me in the direction of a solution? I'm pretty new to firejail, but it seems like a good tool.

**Comment by ghost** [comment, marked as answer]
There's a few things you'll need to add in a `~/.config/firejail/qutebrowser.local` to support mpv inside the qutebrowser sandbox: access to mpv's configuration and allowing Lua basically. Here's an example:

```
# Firejail profile for qutebrowser
# Persistent local customizations

## allow mpv

noblacklist ${HOME}/.config/mpv
whitelist ${HOME}/.config/mpv

# Allow lua (blacklisted by disable-interpreters.inc)
include allow-lua.inc

whitelist /usr/share/lua
whitelist /usr/share/lua*
whitelist /usr/share/vulkan
```

There might be additional things to consider, but try starting with this and report back if you run into other roadblocks.

**Reply by AblatedSprocket** [reply]
That did it, thank you so much! Those inc files are handy! I think if I had known about them, I may have been able to stumble through the rest. I'll remember them for future sandboxing.
